apps/balancer-tools/src/app/apr/(utils)/calculatePoolStats.ts

```diff
@@ -28,6 +30,21 @@ const getDataFromCacheOrCompute = async <T>(
   return computedData;
 };
 
+const getTVLData = async (poolId: string, round: Round) => {
+  const tokenAmount = await getTokenAmountByPoolTimestamp(
+    poolId,
+    round.endDate.getTime() / 1000,
+  );
+
+  const coinPrice = await getPoolTokenPriceByRound(poolId, round);
+
+  return tokenAmount.reduce(
+    (sum, value1, index) =>
+      sum + parseFloat(value1) * parseFloat(coinPrice[index]),
+    0,
+  );
+};
```

Comment:
I think this makes sense for us to get more precise with current round -- but as mentioned below and in [a Slack message](https://bleu-llc.slack.com/archives/C05FE3PLPP0/p1692971277559029) I sent I think it's going to require less API calls and be more consistent with Balancer's own FE to get the historical data from the snapshot